http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/AwaitCloseChannelPoolMap.java

```diff
@@ -123,8 +116,11 @@ public static Builder builder() {
 
     @Override
     protected SimpleChannelPoolAwareChannelPool newPool(URI key) {
-        SslContext sslContext = sslContext(key);
-        
+        SslContext sslContext = null;
+        if (needSslContext(key)) {
+            sslContext = sslContextProvider.sslContext();
+        }
```

Comment:
Nit: ternary, so that sslContext can be effectively-final?

Reply:
+1